hil/ext/switches/ovs.py

```diff
@@ -0,0 +1,271 @@
+from hil.migrations import paths
+from os.path import dirname, join
+logger = logging.getLogger(__name__)
+paths[__name__] = join(dirname(__file__), 'migrations', 'ovs')
+
+
+class VlanError(Exception):
+    """ Raise this exception when vlan cannot be added to the port."""
+    pass
+
+class NotImplementedYet(Exception):
+    """ Raise this exception for functions that require 
+    more information to implement.
+    """
+    pass
+
+class PortInfoNotFound(Exception):
+    """ Raise this exception when this driver is unable to fetch 
+    information for a given port.
+    """
+    pass
+
+class OVS_RequestFailed(Exception):
+    """ Raise this exception when a requestOVS failure
+    cannot does not follow into any of the above exceptions.
+    """
+    pass
+
+class Ovs(Switch):
+    api_name = 'http://schema.massopencloud.org/haas/v0/switches/ovs'
+    dir_name = os.path.dirname(__file__)
+    __mapper_args__ = {
+        'polymorphic_identity': api_name,
+    }
+
+    id = db.Column(
+            BigIntegerType, db.ForeignKey('switch.id'), primary_key=True
+            )
+    hostname = db.Column(db.String, nullable=False)
+    username = db.Column(db.String, nullable=False)
+    password = db.Column(db.String, nullable=False)
+
+    @staticmethod
+    def validate(kwargs):
+        schema.Schema({
+            'username': basestring,
+            'hostname': basestring,
+            'password': basestring,
+        }).validate(kwargs)
+
+    def session(self):
+        return self
+
+    def disconnect(self):
+        pass
+
+    @staticmethod
+    def validate_port_name(port):
+        """ This driver accepts any string as port name."""
+        pass
+
+    def str2list(self, a_string):
+        """ Converts a string representation of list to list.
+        Empty list is put as None.
+        """
+        if a_string[0] == '[' and a_string[1] == ']':
+            a_list = ast.literal_eval(a_string)
+            return a_list
+        elif a_string[0] == '[' and len(a_string) > 2:
+            a_string = a_string.replace("[", "['").replace("]", "']")
+            a_string = a_string.replace(",", "','")
+            a_list = ast.literal_eval(a_string)
+            a_list = [ ele.strip() for ele in a_list]
+            return a_list
+        else:
+            return None
+
+    def str2dict(self, a_string):
+        """Converts a string representation of dictionary to a dictionary."""
+        if a_string[0] == '{' and a_string[1] == '}':
+            a_dict = ast.literal_eval(a_string)
+            return a_dict
+        elif a_string[0] == '{' and len(a_string) > 2:
+            a_string = a_string.replace("{", "{'").replace("}", "'}")
+            a_string = a_string.replace(":", "':'").replace(",", "','")
+            a_dict = ast.literal_eval(a_string)
+            a_dict = {k.strip():v.strip() for k, v in a_dict.iteritems()}
+            return a_dict
+        else:
+            return None
+
+    def _create_session(self):
```

Comment:
Why did you create this new method here, but put a `pass` in the `session` method as defined in the super class?

Reply:
Once you use `session`, all the methods have to go under it. That would mean more indented code, less space per line. As it is I am struggling to work with the 79 character limitation.

Anyways, I later realized that it might not be needed at all. Since `_requestOVS` does most of the heavy lifting.

so will remove `_create_session` that method in the next commit. 
Thanks for pointing it out.

Reply:
> Once you use session, all the methods have to go under it. That would mean more indented code, less space per line

Yes and No. The `session` method will return an object satisfying the interface specified by `SwitchSession`. And the class for that object will have the methods, so it would only be just 1 level of indentation. The [nexus driver](https://github.com/CCI-MOC/hil/blob/master/hil/ext/switches/nexus.py#L84) shows how it's done. 

But since you removed it anyway, I am posting this comment for clarification.